src/include/cephfs/libcephfs.h

```diff
@@ -789,6 +848,20 @@ int ceph_rename(struct ceph_mount_info *cmount, const char *from, const char *to
 int ceph_fstatx(struct ceph_mount_info *cmount, int fd, struct ceph_statx *stx,
 		unsigned int want, unsigned int flags);
 
+/**
+ * Get attributes of a file relative to a file descriptor
+ *
+ * @param cmount the ceph mount handle to use for performing the stat.
+ * @param dirfd open file descriptor (or AT_FDCWD)
+ * @param relpath to the file/directory to get statistics of
+ * @param stx the ceph_statx struct that will be filled in with the file's statistics.
+ * @param want bitfield of CEPH_STATX_* flags showing designed attributes
+ * @param flags bitfield that can be used to set AT_* modifier flags (only AT_NO_ATTR_SYNC and AT_SYMLINK_NOFOLLOW)
+ * @returns 0 on success or negative error code on failure.
+ */
+int ceph_fstatxat(struct ceph_mount_info *cmount, int dirfd, const char *relpath,
```

Comment:
```suggestion
int ceph_statxat(struct ceph_mount_info *cmount, int dirfd, const char *relpath,
```

?

Reply:
There is fstatat(2).

Reply:
Right, but the analog here is `statat` not `fstat`. `fstatat` makes no sense since `fstat` is taking a file descriptor for the file to be stat'd.